src/Umbraco.Core/Security/MembershipProviderExtensions.cs

```diff
@@ -37,11 +37,15 @@ internal static bool CanResetPassword(this MembershipProvider provider, IUserSer
                 var identity = Thread.CurrentPrincipal.GetUmbracoIdentity();
                 if (identity != null)
                 {
-                    var user = userService.GetByUsername(identity.Username);
-                    var userIsAdmin = user.IsAdmin();
-                    if (userIsAdmin)
+                    var userId = 0;
+                    if(int.TryParse(identity.Id.ToString(), out userId))
                     {
-                        canReset = true;
+                        var user = userService.GetUserById(userId);
+                        var userIsAdmin = user.IsAdmin();
```

Comment:
I guess the additional check required here which was the original problem is that there is no null check performed on the `user` variable returned. I'll update and merge.